ast.c

```diff
@@ -560,7 +572,9 @@ static inline zend_ast_attr ast_assign_op_to_binary_op(zend_ast_attr attr) {
 static inline zend_ast **ast_get_children(zend_ast *ast, uint32_t *count) {
 	if (ast_kind_is_decl(ast->kind)) {
 		zend_ast_decl *decl = (zend_ast_decl *) ast;
-#if PHP_VERSION_ID >= 80000
+#if PHP_VERSION_ID >= 80100
+		*count = decl->kind == ZEND_AST_CLASS ? (decl->child[4] != NULL ? 5 : 4) : 5;
```

Comment:
I think we don't have anything like this right now (where an AST node has an optional child). Maybe it's better to unconditionally include child 4 in the new version?